homeassistant/components/landisgyr_heat_meter/sensor.py

```diff
@@ -56,38 +76,52 @@ def __init__(self, coordinator, description, device):
         self.entity_description = description
 
         self._attr_device_info = device
-        self._attr_should_poll = bool(self.key in ("heat_usage", "heat_previous_year"))
-
-    async def async_added_to_hass(self) -> None:
-        """Call when entity about to be added to hass."""
-        await super().async_added_to_hass()
-        state = await self.async_get_last_sensor_data()
-        if state:
-            self._attr_native_value = state.native_value
+        self.update_values_from_api()
+
+    def update_values_from_api(self):
+        """Update attribute values with the data from the api coordinator."""
+        if self.coordinator.data is None:
+            _LOGGER.debug(
+                "Cannot update %s: could not read any data from the Heat Meter",
+                self.key,
+            )
+            return
 
-    def _handle_coordinator_update(self) -> None:
-        """Handle updated data from the coordinator."""
-        if self.key in asdict(self.coordinator.data):
+        api_key = API_KEY[self.key]
+        if api_key in asdict(self.coordinator.data):
             if self.device_class == SensorDeviceClass.TIMESTAMP:
                 self._attr_native_value = dt_util.as_utc(
-                    asdict(self.coordinator.data)[self.key]
+                    asdict(self.coordinator.data)[api_key]
                 )
             else:
-                self._attr_native_value = asdict(self.coordinator.data)[self.key]
-
-        if self.key == "heat_usage":
-            self._attr_native_value = convert_gj_to_mwh(
-                self.coordinator.data.heat_usage_gj
-            )
+                self._attr_native_value = asdict(self.coordinator.data)[api_key]
 
-        if self.key == "heat_previous_year":
-            self._attr_native_value = convert_gj_to_mwh(
-                self.coordinator.data.heat_previous_year_gj
-            )
+        _LOGGER.debug(
+            "Updated value of %s",
+            self.key,
+        )
```

Comment:
Home Assistant already logs entity state changes in debug mode, no need to add another one.